src/System.Private.CoreLib/src/System/Diagnostics/Eventing/EventPipeController.cs

```diff
@@ -90,6 +90,7 @@ internal static void Initialize()
                         s_controllerInstance = new EventPipeController();
                     }
                     // If enable is explicitly set to 0, then don't start the controller (to avoid overhead).
+                    RuntimeEventSource.m_RuntimeEventSource = new RuntimeEventSource();
```

Comment:
Instead of this, add a static `Initialize` method on RuntimeEventSource() that will create the singleton and will be called from here.

Keep the `m_RuntimeEventSource` field private.